### PR TITLE
test: renovate golden table implementation

### PR DESCRIPTION
**Pull request by gaissmai**
## Summary by CodeRabbit

- **Tests**
  - Updated table comparison coverage for deletion, modification, union, lookup, aggregation, and overlapping-prefix scenarios.
  - Improved validation for key-only and complete table results across IPv4, IPv6, and mixed-address cases.
  - Added broader parallelized coverage for table operations and edge cases.
  - Updated golden test helpers and benchmarks to reflect current table behavior.

- **Chores**
  - Refreshed aggregation benchmarks and test utilities.
  - No user-facing functionality or public API changes.

### CODE REVIEW

**Comment by gaissmai** [comment]
> and single-prefix aggregation retains a value it should clear.

@coderabbit den Satz verstehe ich nicht, zeige mir die zeile im source code dazu

File: internal/tests/golden/table.go [hunk 1]
```diff
@@ -20,68 +20,113 @@ func cmpPrefix(a, b netip.Prefix) int {
...
+	}
+
+	slices.SortFunc(result, cmpPrefix)
+	return result
+}
+
+// Equal reports whether ta and tb contain the exact same set of key-value pairs.
+// Values are compared using equality (==) via dynamic interface boxing.
+func (ta Table[V]) Equal(tb Table[V]) bool {
+	return maps.EqualFunc(ta, tb, func(v1, v2 V) bool { return any(v1) == any(v2) })
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/tests/golden/table.go:75]: ich werde diese Lösung beibehalten wie ich sie habe, das ist eine bewusste Entscheidung, denn dies ist nur eine golden test table

File: internal/tests/golden/table_test.go [hunk 2]
```diff
@@ -384,46 +561,44 @@ func TestTableAll(t *testing.T) {
...
-
-			if !slices.Equal(got, tt.want) {
-				t.Errorf("All() = %v, want %v", got, tt.want)
+				wantVal, ok := tt.want[pfx]
+				if !ok {
+					t.Errorf("All(): not ok, missing val %v", val)
+				} else if val != wantVal {
+					t.Errorf("All(): want: %v, got: %v", wantVal, val)
+				}
 			}
```
[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/tests/golden/table_test.go:727]: ich finde die Stelle nicht in der Datei unter der Zeilennummer

[the remark this answers is not in the document]
Comment (gaissmai) [reply internal/tests/golden/table_test.go:727]: ich pushe demnächst noch einmal, dann nimm bitte die richtige Zeilennummer

**Comment by gaissmai** [comment]
@coderabbit pause
@greptileai pause

**Comment by gaissmai** [comment]
@coderabbit resume
@greptileai resume
